include/roboptim/core/function.hh

```diff
@@ -47,6 +47,18 @@
 
 # include <roboptim/core/detail/utility.hh>
 
+// We should probably move these templates out of function.hh
+// Flags used for type checking inside the RobOptim function architecture
+# define ROBOPTIM_IS_DIFFERENTIABLE 1UL << 0
+# define ROBOPTIM_IS_TWICE_DIFFERENTIABLE 1UL << 1
+# define ROBOPTIM_IS_LINEAR 1UL << 2
+# define ROBOPTIM_IS_NUMERIC_LINEAR 1UL << 3
+# define ROBOPTIM_IS_QUADRATIC 1UL << 4
+# define ROBOPTIM_IS_NUMERIC_QUADRATIC 1UL << 5
+# define ROBOPTIM_IS_CONSTANT 1UL << 6
+# define ROBOPTIM_IS_POLYNOMIAL 1UL << 7
```

Comment:
The problem with the enum is that we therefore need to properly define the requested values (for example the one roboptim-core-manifold will use to reecognize a functionOnManifold).
So, shall we add all the module-dependant entries directly into roboptim-core (bad idea I guess) or just define 16 reserved (available) entries and make clear every programmer that wants to add feature must use one of them.

Another possibility (the cleaner for roboptim-core) could be to force the modules to access and modify it bit-wise (only using the enumeration for it is defined, RobOptim types). It would be a mix of both solutions in each module